samples/java-spring-cinema-booking-choreography/src/main/java/com/example/cinema/model/CinemaApiModel.java

```diff
@@ -0,0 +1,65 @@
+package com.example.cinema.model;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.util.List;
+
+public interface CinemaApiModel {
+    sealed interface ShowCommand {
+
+      record CreateShow(String title, int maxSeats) implements ShowCommand {}
+
+      record ReserveSeat(String walletId, String reservationId, int seatNumber) implements ShowCommand {}
+
+      record ConfirmReservationPayment(String reservationId) implements ShowCommand {}
+
+          record CancelSeatReservation(String reservationId) implements ShowCommand {}
+    }
+
+    enum ShowCommandError {
+      SHOW_ALREADY_EXISTS,
+      SHOW_NOT_FOUND,
+      TOO_MANY_SEATS,
+      SEAT_NOT_FOUND,
+      SEAT_NOT_AVAILABLE,
+      RESERVATION_NOT_FOUND,
+      DUPLICATED_COMMAND,
+      CANCELLING_CONFIRMED_RESERVATION
+    }
+
+    @JsonTypeInfo(use = JsonTypeInfo.Id.NAME)
+    @JsonSubTypes({
+      @JsonSubTypes.Type(value = Response.Success.class),
+      @JsonSubTypes.Type(value = Response.Failure.class)
+    })
```

Comment:
I do think there is the possibility to remove this annotations if we use `sealed` trait. Maybe something to add to the ObjectMapper. We need to check it.

Reply:
Not really, the issue is still open: https://github.com/FasterXML/jackson-module-scala/issues/289